tmail-backend/jmap/extensions-es-v7/src/main/java/com/linagora/tmail/james/jmap/ESEmailAddressContactSearchEngine.java

```diff
@@ -120,6 +124,76 @@ public Publisher<EmailAddressContact> autoComplete(AccountId accountId, String p
+    }
+
+    @Override
+    public Publisher<EmailAddressContact> list(Domain domain) {
+        SearchRequest request = new SearchRequest(configuration.getDomainContactReadAliasName().getValue())
+            .source(new SearchSourceBuilder()
+                .query(QueryBuilders.boolQuery()
+                    .should(QueryBuilders.termQuery(DOMAIN, domain.asString()))
+                    .minimumShouldMatch(1)));
+
+        return client.search(request, RequestOptions.DEFAULT)
+            .flatMapIterable(searchResponse -> ImmutableList.copyOf(searchResponse.getHits().getHits()))
+            .map(Throwing.function(this::extractContentFromHit).sneakyThrow());
+    }
+
+    @Override
+    public Publisher<EmailAddressContact> listDomainsContacts() {
+        SearchRequest request = new SearchRequest(configuration.getDomainContactReadAliasName().getValue())
+            .source(new SearchSourceBuilder().query(QueryBuilders.matchAllQuery()));
+
+        return client.search(request, RequestOptions.DEFAULT)
+            .flatMapIterable(searchResponse -> ImmutableList.copyOf(searchResponse.getHits().getHits()))
+            .map(Throwing.function(this::extractContentFromHit).sneakyThrow());
+    }
+
+    @Override
+    public Publisher<EmailAddressContact> get(AccountId accountId, MailAddress mailAddress) {
+        return client.get(new GetRequest(configuration.getUserContactReadAliasName().getValue())
+                    .id(computeUserContactDocumentId(accountId, mailAddress).asString())
+                    .routing(mailAddress.asString()),
+                RequestOptions.DEFAULT)
+            .filter(GetResponse::isExists)
+            .map(GetResponse::getSourceAsMap)
+            .map(Throwing.function(this::extractContactFromSource).sneakyThrow())
+            .switchIfEmpty(Mono.error(new ContactNotFoundException(mailAddress)));
+    }
+
+    @Override
+    public Publisher<EmailAddressContact> get(Domain domain, MailAddress mailAddress) {
+        return client.get(new GetRequest(configuration.getDomainContactReadAliasName().getValue())
+                    .id(computeDomainContactDocumentId(domain, mailAddress).asString())
+                    .routing(mailAddress.asString()),
+                RequestOptions.DEFAULT)
+            .filter(GetResponse::isExists)
+            .map(GetResponse::getSourceAsMap)
+            .map(Throwing.function(this::extractContactFromSource).sneakyThrow())
+            .switchIfEmpty(Mono.error(new ContactNotFoundException(mailAddress)));
+    }
+
+    private EmailAddressContact extractContactFromSource(Map<String, Object> source) throws AddressException {
+        return new EmailAddressContact(
+            UUID.fromString((String) source.get(CONTACT_ID)),
+            new ContactFields(
+                new MailAddress((String) source.get(EMAIL)),
+                (String) source.get(FIRSTNAME),
+                (String) source.get(SURNAME)
+            ));
+    }
+
```

Comment:
Outside of this range but please avoid duplicating calls to `getSourceAsMap` within `extractContentFromHit`.

Do we have tests for contacts autocomplete without FIRSTNAME SURNAME ?

Reply:
> Do we have tests for contacts autocomplete without FIRSTNAME SURNAME ?

There is unit tests regarding it. Do you want some for the jmap method tests as well or that's enough?